bundles/org.openhab.binding.ring/src/main/java/org/openhab/binding/ring/internal/RingVideoServlet.java

```diff
@@ -43,72 +50,107 @@
  * @author Peter Mietlowski (zolakk) - Initial contribution
  * @author Ben Rosenblum - Updated for OH4 / New Maintainer
  */
-@Component(service = HttpServlet.class)
+@Component(service = { Servlet.class, RingVideoServlet.class }, immediate = true)
+@HttpWhiteboardServletName(SERVLET_VIDEO_PATH)
+@HttpWhiteboardServletPattern({ SERVLET_VIDEO_PATH, SERVLET_VIDEO_PATH + "/*" })
 @NonNullByDefault
 public class RingVideoServlet extends HttpServlet {
-
     private static final long serialVersionUID = -5592161948589682812L;
 
     private final Logger logger = LoggerFactory.getLogger(RingVideoServlet.class);
 
-    private String videoStoragePath = "";
+    private final Map<ThingUID, Path> pathRegistrations = new ConcurrentHashMap<>();
+    private List<Path> videoPaths = List.of();
 
     public RingVideoServlet() {
     }
 
-    public RingVideoServlet(HttpService httpService, String videoStoragePath) {
-        Path path = Paths.get(videoStoragePath);
-        FileSystem fs = path.getFileSystem();
-        String sep = fs.getSeparator();
-        this.videoStoragePath = videoStoragePath + (videoStoragePath.endsWith(sep) ? "" : sep);
-        try {
-            httpService.registerServlet(SERVLET_VIDEO_PATH, this, null, httpService.createDefaultHttpContext());
-        } catch (NamespaceException | ServletException e) {
-            logger.warn("Register servlet fails", e);
-        }
+    public void addVideoStoragePath(ThingUID thingUID, String pathString) {
+        Path path = Paths.get(pathString);
+        pathRegistrations.put(thingUID, path);
+        determineAllVideoPaths();
+    }
+
+    public void removeVideoStoragePath(ThingUID thingUID) {
+        pathRegistrations.remove(thingUID);
+        determineAllVideoPaths();
+    }
+
+    private synchronized void determineAllVideoPaths() {
+        videoPaths = pathRegistrations.values().stream().distinct().toList();
     }
 
-    @SuppressWarnings("null")
     @Override
-    protected void service(HttpServletRequest request, HttpServletResponse response)
-            throws ServletException, IOException {
-        String ipAddress = request.getHeader("HTTP_X_FORWARDED_FOR");
-        if (ipAddress == null) {
-            ipAddress = request.getRemoteAddr();
+    protected void doGet(@NonNullByDefault({}) HttpServletRequest request,
+            @NonNullByDefault({}) HttpServletResponse response) throws ServletException, IOException {
+        String requestURI = request.getRequestURI();
+
+        if (SERVLET_VIDEO_PATH.equals(requestURI)) {
+            String responseString = """
+                    <html lang="en-US">
+                    <head>
+                    <title>Ring - Video files</title>
+                    </head>
+                    <body>
+                    <h1>Available video files:</h1>
+                    """;
+
+            responseString += videoPaths.stream().flatMap(path -> {
+                try {
+                    return Files.list(path);
```

Comment:
The try-with-resources construct should be used to ensure that the stream's `close` method is invoked after the stream operations are completed.